tidy3d/components/data/sim_data.py

```diff
@@ -792,6 +792,28 @@ def plot_scalar_array(
 
         return ax
 
+    def to_mat(self, fname):
+        """Output the SimulationData object as .mat MATLAB file
+
+        Parameters
+        ----------
+        fname : str
+            Full path to the output file. Should include .mat file extension
+
+        Example
+        -------
+        >>> simData.to_mat('/path/to/file/data.mat') # doctest: +SKIP
+        """
+        # Get SimData object as dictionary
+        simDict = self.dict()
+
+        # Remove NoneType values from dict
+        # Built from theory discussed in https://github.com/scipy/scipy/issues/3488
+        modifiedSimDict = replace_values(simDict, None, [])
+
+        # Savemat hasn't actually implemented long_field_names, but set as true in case someone notices the bug
+        savemat(fname, modifiedSimDict, long_field_names=True)
```

Comment:
because this is somewhat of an experimental feature, I would say we might want to do something like
```py
try:
    savemat(...)
except Exception as e:
    raise ValueError(f"Could not save supplied 'SimulationData' to file. As this is an experimental feature, we may not be able to support the contents of your dataset. If you receive this error, please feel free to raise an issue on our front end repository so we can investigate.") from e
```

which will give a nicer error and then provide the original stack exchange (because of that `from e` bit).